src/xtgeo/grid3d/_gridprop_import_grdecl.py

```diff
@@ -83,6 +80,97 @@ def import_bgrdecl_prop(self, pfile, name="unknown", grid=None):
     pfile.cfclose()
 
 
+@contextmanager
+def open_grdecl(grdecl_file, keywords):
+    """
+    Opens the grdecl_file for reading. Is a generator for tuples of keyword /
+    values in records of that file. The format of the file must be that of the
+    GRID section of a eclipse input DATA file.
+
+    The records looked for must be "simple" ie.  start with the keyword, be
+    followed by single word values and ended by a slash ('/').
+
+    open_grdecl does not follow includes, obey skips, parse MESSAGE commands or
+    make exception for groups and subrecords.
+
+    Raises:
+        ValueError: when end of file is reached without terminating a keyword.
+
+    Args:
+        keywords (List[str]): Which keywords to look for, these are expected
+            to be at the start of a line in the file and be followed by
```

Comment:
`be followed by` by what? :)

Reply:
Keyword is followed by values and ends by slash:
```
keyword
value value value \
```

Reply:
Feel free to help with the description here, the point is that this is a record in the simple form described above.

Reply:
Just, can do. Just note that I didn't criticise the content, just that it was only hanging there. I would expect it to contain what it was followed by as well (or did I not see the rest??)

Reply:
> to be at the start of a line in the file and the respective values following on subsequent lines separated either by whitespace or linebreak. Reading of a keyword is completed by a final \


I actually think it's fair to add an example as well.

Reply:
Used your suggestion and added a code-block with example.